packages/experience-builder-sdk/src/blocks/VisualEditorContext.tsx

```diff
@@ -187,6 +201,33 @@ export function VisualEditorContextProvider({
           }
           break;
         }
+        case INCOMING_EVENTS.DesignComponentsUpdated: {
```

Comment:
This is more of `DesignComponentAdded` event, right? It handles only a single component and it seems it's only applicable when a new component is added to the composition. The sys object and component definition won't change after a component has been created/added so what does 'updated' mean here?
Also is it possible to have this event only update the registry so there is some separation of responsibilities each event has a single responsibility. The tree will be updated by the `INCOMING_EVENTS.CompositionUpdated` anyway right?

Reply:
I don't call the CompositionUpdated event on design component created, only this event. I did this because `CompositionUpdated` also sets datasource, design, values and the rest of them, so I wanted there to be a split. So whenever the tree is updated as a result of design component created and added to the tree, I send the tree and other other payload in this event. 

Also, good point about the naming. Will change